### ISSUE TITLE
Legacy Detection: False positive `GetPhysicalDeviceProperties`

### ISSUE THREAD

**Issue by codecnotsupported** [KhronosGroup/Vulkan-ValidationLayers#11305]
**Environment:**
 - OS: Linux
 - GPU and driver version: Intel B580, Mesa 25.3.1-arch1.2
 - SDK or header version if building from repo: 1.4.335.0
 - Options enabled (synchronization, best practices, etc.): VK_VALIDATION_LEGACY_DETECTION (Legacy Detection)

**Describe the Issue**

The validation's `VK_VALIDATION_LEGACY_DETECTION` gives a false positive due to the validation layer calling `GetPhysicalDeviceProperties`.
```
Validation Warning: [ WARNING-legacy-gpdp2 ] | MessageID = 0xde900250
vkGetPhysicalDeviceProperties(): vkGetPhysicalDeviceProperties is a legacy command and this VkInstance was created with VK_VERSION_1_1 which contains vkGetPhysicalDeviceProperties2 that can be used instead.
See more information about this legacy in the specification: https://docs.vulkan.org/spec/latest/appendices/legacy.html#legacy-gpdp2
Objects: 1
    [0] VkPhysicalDevice 0x5603a2fa58c0
```

`#7  0x00007ffff36f26b9 in vulkan_layer_chassis::GetPhysicalDeviceProperties (physicalDevice=0x555555883370, pProperties=0x7fffffffc040) at /vulkan-sdk/1.4.335.0/source/Vulkan-ValidationLayers/layers/vulkan/generated/chassis.cpp:285`

**Expected behavior**

Expected no warning.

**Valid Usage ID**
N/A

**Additional context**

<details>
<summary>Full error</summary>

```log 

Validation Information: [ WARNING-CreateInstance-status-message ] | MessageID = 0x23dfd876
vkCreateInstance(): Khronos Validation Layer Active:
    Current Enables: VK_VALIDATION_LEGACY_DETECTION.
    Current Disables: None.

Objects: 1
    [0] VkInstance 0x5603a2f68300

Validation Warning: [ WARNING-legacy-gpdp2 ] | MessageID = 0xde900250
vkGetPhysicalDeviceProperties(): vkGetPhysicalDeviceProperties is a legacy command and this VkInstance was created with VK_VERSION_1_1 which contains vkGetPhysicalDeviceProperties2 that can be used instead.
See more information about this legacy in the specification: https://docs.vulkan.org/spec/latest/appendices/legacy.html#legacy-gpdp2
Objects: 1
    [0] VkPhysicalDevice 0x5603a2fa58c0
```

</details>

<details>
<summary>Full backtrace</summary>

```
0x00007ffff7c9890c in ?? () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff7c9890c in ?? () from /usr/lib/libc.so.6
#1  0x00007ffff7c3e3a0 in raise () from /usr/lib/libc.so.6
#2  0x00007ffff3fddee2 in MessengerBreakCallback (message_severity=<optimized out>, message_type=<optimized out>, callback_data=<optimized out>, user_data=<optimized out>)
    at /vulkan-sdk/1.4.335.0/source/Vulkan-ValidationLayers/layers/error_message/logging.cpp:722
#3  0x00007ffff3fe2d61 in DebugReport::LogMessage (this=this@entry=0x555555880630, msg_flags=msg_flags@entry=2, vuid_text=..., objects=..., loc=..., 
    main_message="vkGetPhysicalDeviceProperties is a legacy command and this VkInstance was created with VK_VERSION_1_1 which contains vkGetPhysicalDeviceProperties2 that can be used instead.\nSee more information about"...)
    at /vulkan-sdk/1.4.335.0/source/Vulkan-ValidationLayers/layers/error_message/logging.cpp:239
#4  0x00007ffff3fe38db in DebugReport::LogMessageVaList (this=0x555555880630, msg_flags=msg_flags@entry=2, vuid_text="WARNING-legacy-gpdp2", objects=..., loc=..., 
    format=0x7ffff43a92c8 "vkGetPhysicalDeviceProperties is a legacy command and this VkInstance was created with VK_VERSION_1_1 which contains vkGetPhysicalDeviceProperties2 that can be used instead.\nSee more information about"..., 
    argptr=0x7fffffffbd60) at /vulkan-sdk/1.4.335.0/source/Vulkan-ValidationLayers/layers/error_message/logging.cpp:711
#5  0x00007ffff334d6c1 in Logger::LogWarning (this=<optimized out>, vuid_text=..., objlist=..., loc=..., 
    format=format@entry=0x7ffff43a92c8 "vkGetPhysicalDeviceProperties is a legacy command and this VkInstance was created with VK_VERSION_1_1 which contains vkGetPhysicalDeviceProperties2 that can be used instead.\nSee more information about"...)
    at /vulkan-sdk/1.4.335.0/source/Vulkan-ValidationLayers/layers/./error_message/logging.h:308
#6  0x00007ffff37e4dbd in legacy::Instance::PreCallValidateGetPhysicalDeviceProperties (this=<optimized out>, physicalDevice=<optimized out>, pProperties=<optimized out>, error_obj=...) at /usr/include/c++/11/bits/char_traits.h:399
#7  0x00007ffff36f26b9 in vulkan_layer_chassis::GetPhysicalDeviceProperties (physicalDevice=0x555555883370, pProperties=0x7fffffffc040) at /vulkan-sdk/1.4.335.0/source/Vulkan-ValidationLayers/layers/vulkan/generated/chassis.cpp:285
#8  0x00007ffff64fa0d4 in ?? () from /usr/lib/libVkLayer_MESA_device_select.so
#9  0x00007ffff64f81a7 in ?? () from /usr/lib/libVkLayer_MESA_device_select.so
#10 0x00007ffff64f8676 in ?? () from /usr/lib/libVkLayer_MESA_device_select.so
#11 0x00007ffff64f9dd0 in ?? () from /usr/lib/libVkLayer_MESA_device_select.so
#12 0x00007ffff7ef50e1 in vkEnumeratePhysicalDevices (instance=<optimized out>, pPhysicalDeviceCount=0x7fffffffc774, pPhysicalDevices=0x0) at /vulkan-sdk/1.4.335.0/source/Vulkan-Loader/loader/trampoline.c:866
#13 0x000055555569d8a5 in ash::instance::{impl#3}::enumerate_physical_devices::{closure#0} (count=0x7fffffffc774, data=0x0) at ~/.cargo/git/checkouts/ash-b4604afb699e89f5/c9292cf/ash/src/instance.rs:496
#14 0x000055555569d4da in ash::read_into_uninitialized_vector<u32, ash::vk::definitions::PhysicalDevice, ash::instance::{impl#3}::enumerate_physical_devices::{closure_env#0}> (f=...)
    at ~/.cargo/git/checkouts/ash-b4604afb699e89f5/c9292cf/ash/src/lib.rs:258
#15 0x000055555569cd67 in ash::instance::Instance::enumerate_physical_devices (self=0x7fffffffcf70) at ~/.cargo/git/checkouts/ash-b4604afb699e89f5/c9292cf/ash/src/instance.rs:495
#16 0x000055555569c829 in graphics_engine::get_physical_device (instance=0x7fffffffcf70) 
#17 0x000055555569c612 in graphics_engine::create_device (instance=0x7fffffffcf70) 
#18 0x000055555569ca67 in graphics_engine::init () 
```

</details>

**Comment by spencer-lunarg** [comment]
@codecnotsupported I don't think this is a false positive

https://docs.vulkan.org/spec/latest/appendices/legacy.html#legacy-gpdp2 (and the `vk.xml`) clearly has marked `vkGetPhysicalDeviceProperties` as legacy

looking at your stacktrace, the issue is `libVkLayer_MESA_device_select.so` is the one calling it, as you can see in the [source code](https://gitlab.freedesktop.org/mesa/mesa/-/blob/main/src/vulkan/device-select-layer/device_select_layer.c?ref_type=heads#L197)

The Validation Layers, as a layer, has zero way to detect if the call is coming from the application or another layer

If you want, you can set `export NODEVICE_SELECT=0` and that will disable the `VK_LAYER_MESA_device_select` layer and that will remove the warning being triggered

**Comment by codecnotsupported** [comment]
Thanks you were right.
Ran with `NODEVICE_SELECT=0`, warning went away. 
It was indeed `libVkLayer_MESA_device_select.so`, closing.
